packages/flutter/lib/src/widgets/selectable_region.dart

```diff
@@ -2457,6 +2481,76 @@ abstract class MultiSelectableSelectionContainerDelegate extends SelectionContai
     );
   }
 
+  /// This method calculates a local [SelectedContentRange] based on the list 
+  /// of [ranges] that are accumulated from the [Selectable] children under this
+  /// delegate. This calculation takes into account the accumulated content
+  /// length before the active selection, and returns a [SelectedContentRange.empty]
+  /// when either selection edge has not been set.
+  SelectedContentRange _calculateLocalRange(List<SelectedContentRange> ranges) {
+    // Calculate content length.
+    int totalContentLength = 0;
+    for (final SelectedContentRange range in ranges) {
+      totalContentLength += range.contentLength;
+    }
+    if (currentSelectionStartIndex == -1 || currentSelectionEndIndex == -1) {
+      return SelectedContentRange.empty(contentLength: totalContentLength);
+    }
+    int startOffset = 0;
+    int endOffset = 0;
+    bool foundStart = false;
+    bool forwardSelection = currentSelectionEndIndex >= currentSelectionStartIndex;
+    if (currentSelectionEndIndex == currentSelectionStartIndex) {
+      // Determining selection direction is innacurate if currentSelectionStartIndex == currentSelectionEndIndex.
+      // Use the range from the selectable within the selection as the source of truth for selection direction.
+      final SelectedContentRange rangeAtSelectableInSelection = selectables[currentSelectionStartIndex].getSelection();
+      forwardSelection = rangeAtSelectableInSelection.endOffset >= rangeAtSelectableInSelection.startOffset;
```

Comment:
They are not if we have LTR mixed with RTL. We only use the `currentSelectionStartIndex` when `currentSelectionStartIndex == currentSelectionEndIndex`, i.e. there's only one selectable in the selection so that can be thought of as the source of truth for the selection direction. When `currentSelectionStartIndex` and `currentSelectionEndIndex` are different (multiple selectables are under the selection), then we can use `currentSelectionEndIndex >= currentSelectionStartIndex` as the indicator for selection direction.